OutOfSchool/OutOfSchool.WebApi/Controllers/V1/PersonalInfoController.cs

```diff
@@ -42,17 +42,14 @@ public class PersonalInfoController : ControllerBase
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     public async Task<IActionResult> GetPersonalInfo()
     {
-        ShortUserDto info;
+        var info = null as ShortUserDto;
+
         if (currentUserService.IsInRole(Role.Parent))
         {
             info = await parentService.GetPersonalInfoByUserId(currentUserService.UserId);
         }
-        else
-        {
-            info = await userService.GetById(currentUserService.UserId);
-        }
 
-        return info is null ? NoContent() : Ok(info);
+        return Ok(info ?? (await userService.GetById(currentUserService.UserId)));
```

Comment:
(Logic) You've slightly changed logic of this method for case when `parentService.GetPersonalInfoByUserId` returns `null`. I'm not sure if it was intended. Please clarify.

Reply:
In our recent daily meeting, we discussed the importance of personal information for the proper functioning of the app’s frontend. As part of our registration logic update, we now create users without parents initially, allowing them to complete their registration later. However, we need to adjust this process. When a user in the parent role, who is not fully registered, attempts to retrieve their personal information, the current endpoint returns a 403 forbidden error. This behavior is incorrect, so I removed the call to the `HasRights` method in the parent service. Additionally, the frontend assumes that all users have some related information (such as name and role), even if they are not fully registered.

Reply:
Thanks for explanation!